runtime/drivers/env.go

```diff
@@ -0,0 +1,102 @@
+package drivers
+
+import (
+	"context"
+	"database/sql/driver"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/go-yaml/yaml"
+)
+
+// EnviornmentVariables holds the env variables set at an instance level
+// first level in nested map will correspond to kind of env variable like user defined env, secrets, global etc
+// second level will be name of the key
+// Using a map in place of struct at first level, in order to be able to refer keys by small letters
+// for eg : {{.env.timeout}} instead of {{.Env.timeout}}
+// This places a constraint that env variables need to be flattened
+type EnviornmentVariables map[string]map[string]string
+
+// NewEnvVariables creates a new EnviornmentVariables.
+// Defaults are picked from yamlFile
+// envString correspond to user set values
+func NewEnvVariables(ctx context.Context, yamlFile, envString string) (EnviornmentVariables, error) {
```

Comment:
In light of refactors suggested elsewhere, the logic encapsulated here will probably need to be split out into other packages